dataflow/snippets/src/test/java/com/example/dataflow/ApacheIcebergIT.java

```diff
@@ -0,0 +1,176 @@
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.data.GenericRecord;
+import org.apache.iceberg.data.IcebergGenerics;
+import org.apache.iceberg.data.Record;
+import org.apache.iceberg.data.parquet.GenericParquetWriter;
+import org.apache.iceberg.hadoop.HadoopInputFile;
+import org.apache.iceberg.hadoop.HadoopOutputFile;
+import org.apache.iceberg.io.CloseableIterable;
+import org.apache.iceberg.io.FileAppender;
+import org.apache.iceberg.parquet.Parquet;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.types.Types.NestedField;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class ApacheIcebergIT {
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  @Rule
+  public TemporaryFolder temporaryFolder = new TemporaryFolder();
+  private String location;
+
+  private Configuration hadoopConf = new Configuration();
+  private Catalog catalog;
+  private static final TableIdentifier tableId = TableIdentifier.of("db", "table1");
+  private Table table;
+
+  private static final String ouputFileName = "output-00000-of-00001.txt";
+
+  private void createIcebergTable(Catalog catalog, TableIdentifier tableId) throws IOException {
+    var schema = new Schema(
+        NestedField.required(1, "id", Types.LongType.get()),
+        NestedField.optional(2, "name", Types.StringType.get()));
+
+    table = catalog.createTable(tableId, schema);
+  }
+
+  private void writeRecords()
+      throws IOException {
+    GenericRecord record = GenericRecord.create(table.schema());
+    ImmutableList<Record> records =
+        ImmutableList.of(
+            record.copy(ImmutableMap.of("id", 0L, "name", "Person-0")),
+            record.copy(ImmutableMap.of("id", 1L, "name", "Person-1")),
+            record.copy(ImmutableMap.of("id", 2L, "name", "Person-2")));
+
+    Path path = new Path(location, "file1.parquet");
+
+    FileAppender<Record> appender =
+        Parquet.write(HadoopOutputFile.fromPath(path, hadoopConf))
+            .createWriterFunc(GenericParquetWriter::buildWriter)
+            .schema(table.schema())
+            .overwrite()
+            .build();
+    appender.addAll(records);
+    appender.close();
+
+    DataFile dataFile = DataFiles.builder(PartitionSpec.unpartitioned())
+        .withInputFile(HadoopInputFile.fromPath(path, hadoopConf))
+        .withMetrics(appender.metrics())
+        .build();
+
+    table.newFastAppend()
+        .appendFile(dataFile)
+        .commit();
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+
+    // Create an Apache Iceberg catalog with a table and write some test data.
+    File warehouseFolder = temporaryFolder.newFolder();
```

Comment:
nit: I am unfamiliar with this package. However, reading the code which stores a result of the call to `newFolder` in a `File` object looks strange to me. What is a benefit of using `org.junit.rules.TemporaryFolder` instead of standard JDK methods for creating temp file?

Reply:
I think the advantage of the junit API is it deletes the folder on cleanup, but regular temp folder works too. (Fixed)